packages/gatsby/src/utils/client-assets-for-template.ts

```diff
@@ -98,14 +97,24 @@ export async function getScriptsAndStylesForTemplate(
     //     }
     //   }
     // }
+
     const childAssets = webpackStats.childAssetsByChunkName[chunkName]
     if (!childAssets) {
       continue
     }
 
     for (const [rel, assets] of Object.entries(childAssets)) {
+      let childNonTemplateJSAssets
+      // Remove JS asset for templates
+      if (rel !== `app`) {
```

Comment:
rel would be `preload` or `prefetch`, we should check `chunkName` here

Reply:
👍